internals/overlord/restart/restart.go

```diff
@@ -60,7 +61,7 @@ type restartManagerKey struct{}
 // RestartManager takes care of restart-related state.
 type RestartManager struct {
 	state      *state.State
-	restarting RestartType
+	restarting atomic.Int32 // really of type RestartType
```

Comment:
In future we could use an `atomic.Pointer[RestartType]` instead, although unnecessary heap allocation for such a small type, keeps the type.